libafl_nyx/src/executor.rs

```diff
@@ -0,0 +1,185 @@
+use std::{
+    fmt::{self, Debug},
+    marker::PhantomData,
+    path::{PathBuf},
+};
+
+
+use libafl::{
+    executors::{Executor, ExitKind, HasObservers},
+    inputs::{HasBytesVec, Input},
+    observers::ObserversTuple,
+};
+use libnyx::{NyxProcess, NyxReturnValue};
+
+pub type NyxResult<T> = Result<T, String>;
+
+pub struct NyxHelper {
+    nyx_process: NyxProcess,
+    real_map_size: usize,
+    pub map_size: usize,
+    //shared memory with instruction bitmaps
+    pub trace_bits: *mut u8,
+}
+
+const MAX_FILE: u32 = 1024 * 1024;
+pub enum NyxProcessType {
+    // stand alone mode
+    ALONE,
+    PARENT,
+    CHILD,
+}
+impl NyxHelper{
+    pub fn new(
+        target_dir: PathBuf,
+        cpu_id: u32,
+        snap_mode: bool,
+        nyx_type: NyxProcessType,
+    ) -> NyxResult<Self> {
+        let sharedir = target_dir.to_str().unwrap();
+        let workdir = target_dir.join("workdir");
+        let workdir = workdir.to_str().unwrap();
+        let mut nyx_process = match nyx_type {
+            NyxProcessType::ALONE => NyxProcess::new(sharedir, workdir, cpu_id, MAX_FILE, true)?,
+            NyxProcessType::PARENT => {
+                NyxProcess::new_parent(sharedir, workdir, cpu_id, MAX_FILE, true)?
+            }
+            NyxProcessType::CHILD => NyxProcess::new_child(sharedir, workdir, cpu_id, MAX_FILE)?,
+        };
+        let real_map_size = nyx_process.bitmap_buffer_size();
+        let map_size = ((real_map_size + 63) >> 6) << 6;
+        let trace_bits = nyx_process.bitmap_buffer_mut().as_mut_ptr();
+        nyx_process.option_set_reload_mode(snap_mode);
+        nyx_process.option_apply();
+        nyx_process.option_set_timeout(2, 0);
```

Comment:
maybe make this timeout configurable with new's parameter
or even make another method that wraps this like 
```
fn new_with_timeout(timeout: Duration)-> Self{
 nyx.set_timeout()
 nyx.option_apply()
 Self::new()
}

Reply:
Add `NyxHelper::set_timeout`. I follow nyx's format(sec`, `usec`/`microsec`) and write some doc.

Reply:
Specifying timeouts as `core::time::Duration` would be more flexible and more in-line with the rest of LibAFL/rust in general